### DISCUSSION TITLE
Feature Request: Dedicated Syslog Module with standard log header / log format options

### DISCUSSION THREAD

**Discussion by SimSama** [zeek/zeek#1907] [Ideas]
Currently, to forward logs from Zeek to an external system, we use rsyslog and the like to tail the log files and ship them externally. The only problem is, its the wild west where every Zeek client forwarding logs will typically use a different log header for the forwarded log.

Can the Zeek team just implement a syslog module that will auto forward logs with a standard log header for parsing purposes?

e.g.
<timestamp> <sensor hostname> <sensor IP> ZeekLog: <log type>: <json body>
<13>May 2 10:43:27 zeek_sensor_hostname 192.168.1.25 ZeekLog: conn {...usual json contents of log file...}

Different log analyzers and SIEMs have to do battle with clients currently because header formats of the logs that get forwarded are error prone and not consistent for parsing. A standard log header would make it easy. 

Let me know what you think.

**Comment by 0xxon** [comment]
Hi,

thank you for your request. As we typically use github discussions for feature requests, I am going to move this over to a discussion.

**Reply by petiepooo** [reply]
This!  There are ways to do this using syslog-ng, but to have it integrated into Zeek would simplify my world.

**Comment by SimSama** [comment]
Thanks, recommended header was the above.
```
<pri><datetime> <sensor or appliance hostname> <sensor or appliance IP> ZeekLog: <log type> <json body or some other default format>
```
Explicitly placing hostname and sensor IP in log contents helps for those using log relays, so we know the true source of a log in a SIEM, etc. Also using a standard format naming, with "ZeekLog:" specified makes it incredibly clear that this log was from a Zeek appliance. You could even add other things such as version, product, etc for added detail.
